rest/model/dns/zone.go

```diff
@@ -111,6 +128,7 @@ type TSIG struct {
 func NewZone(zone string) *Zone {
 	z := Zone{
 		Zone: zone,
+		Tags: map[string]string{},
```

Comment:
`tags` is not a valid field for a new zone in managed DNS.  Can this be removed from the constructor?

Reply:
From a user's perspective it would be expected that after calling a constructor that the entire structure is initialized appropriately. 

If `Tags` is not initialized here the user then tries to add something to it, a `panic` will be generated: https://play.golang.com/p/vVtxeKFaUaZ

I believe this is more idiomatic and user friendly, but I do understand your concerns given the dual functionality that this module needs to support. I'll remove it if you still want me to.

Reply:
I'm with you on this being more user friendly, but this effectively breaks existing functionality as every request that's submitted to managed using a zone initialized by this constructor will contain an invalid field after this PR is merged.  Our API silently ignores extra fields, but this should still be avoided.

How about removing the field from this constructor, and instead adding a new, DDI specific, constructor (ie `NewDDIZone`) that includes initialization of the tags map?

Reply:
I'll just remove it for now and leave a `NewDDIZone()` construction for another time as similar constructors should be probably be done for other objects (Record, Permissions, etc..) that have similar variances. 

Another option might be to add helpers around the tags (`Set`, `Get`, `IsLocal`, etc..) which could hide the need to consider if the attributes are already initialized.

Reply:
Removed the `Tags` and `BlockedTags` from both `Record` and `Zone` constructors.

Reply:
LGTM.  Love the tag helper idea, could even be generalized to work on all resources with tagging functionality (ie ipam objects as well).